x/challenge/abci.go

```diff
@@ -64,30 +64,40 @@ func EndBlocker(ctx sdk.Context, keeper k.Keeper) {
 		}
 
 		// random redundancy index (sp address)
-		var spOperatorAddress string
-		secondarySpAddresses := objectInfo.SecondarySpAddresses
+		var spOperatorId uint32
 
-		redundancyIndex := k.RandomRedundancyIndex(seed, uint64(len(secondarySpAddresses)+1))
+		bucket, found := keeper.StorageKeeper.GetBucketInfo(ctx, objectInfo.BucketName)
+		if !found {
+			continue
```

Comment:
I think it is quite common to get `gvg` according to the `object` info, so can we wrap it as a function? 
Then the code here will be simple.

Reply:
updated, add a new interface for getting gvg by using bucket info